sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RuleDescriptionSection.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.server.rule;
+
+/**
+ * Represents a sub-section of a rule description (What's the risk, Assess the risk, etc.)
+ *
+ * @since 9.5
+ */
+public interface RuleDescriptionSection {
+
+  /**
+   * This class is a placeholder for the supported rule description section key constants.
+   * @since 9.5
+   */
+  class RuleDescriptionSectionKeys {
+    public static final String INTRODUCTION_SECTION_KEY = "introduction";
+    public static final String WHY_IS_THIS_AN_ISSUE_SECTION_KEY = "why_is_it_an_issue";
+    public static final String HOW_TO_FIX_IT_SECTION_KEY = "how_to_fix_it";
+    public static final String RESOURCES_SECTION_KEY = "resources";
+
+    private RuleDescriptionSectionKeys() {
+      throw new IllegalStateException("this class only serves as a placeholder for section key constants, it must not be instantiated.");
+    }
+  }
+
+  String getKey();
+
+  String getHtmlDescription();
```

Comment:
Just a question regarding the key and the content: so the key would be one of the predefined constants in order to correctly identify the section. Would you expect the section title to be part of the html content ? Ex:
```
<h2>Why is it an issue ?</h2>
<ul>
  <li> ... </li>
  <li> ... </li>
</ul>
```
Or would it be easier to have the title as a separate field ?

Reply:
The idea so far was to **not** have the title included in the HTML description and to have the front-end decide on the title, based on the key. So you would not need to pass the title in the API.
The main reasons for that are:
1. we keep the titles consistent
2. it would (for instance) allow SonarLint to have different titles if necessary for space or other reasons.